File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/achievement/AchievementRewardRequestMessage.scala





















// Generated on 05/03/2013 10:34:32
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait AchievementRewardRequestMessage extends MessageDeserializer {
  val messageId = 6377

  protected trait Message extends BaseMessage with Serializable {
    val messageId = AchievementRewardRequestMessage.messageId

    val achievementId: Short

    def serialize(buf: Buffer) {
      buf.writeShort(achievementId)
    }

  }

  protected def create(achievementId: Short): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val achievementId = buf.readShort()

    create(achievementId)
  }

}

object AchievementRewardRequestMessage extends AchievementRewardRequestMessage {
  case class MessageImpl(achievementId: Short) extends Message
  type Target = MessageImpl

  override def create(achievementId: Short) = MessageImpl(achievementId)

  def apply(achievementId: Short) = create(achievementId)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/AchievementRewardRequestMessage.java


// Generated on 05/08/2013 19:37:38
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class AchievementRewardRequestMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6377;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public short achievementId;
    
    public AchievementRewardRequestMessage() { }
    
    public AchievementRewardRequestMessage(short achievementId) {
        this.achievementId = achievementId;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeShort(achievementId);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        achievementId = buf.readShort();
    }
    
}
